pkg/container/kubernetes/client.go

```diff
@@ -1019,6 +1029,19 @@ func configureContainer(
 		WithStdin(attachStdio).
 		WithTTY(false).
 		WithEnv(envVars...)
+
+	// Add container security context if not already present
+	if container.SecurityContext == nil {
+		userID := int64(1000)
+		groupID := int64(1000)
+		container.WithSecurityContext(
+			corev1apply.SecurityContext().
+				WithAllowPrivilegeEscalation(false).
```

Comment:
We able to add defaults for  `privileged` as well?

If `privileged` is set to false, by default `allowPrivilegeEscalation` will also be false, if you wanted to add `allowPrivilegeEscalation` explicitly though, i don't mind.